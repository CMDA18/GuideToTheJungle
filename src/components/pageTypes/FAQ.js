// @flow
import React, { Component } from 'react'
import Helmet from 'react-helmet'
import styled from 'styled-components'
import Navigation from '../Navigation'
import MainContentGrid from '../MainContentGrid'
import PageTitle from '../PageTitle'
import Intro from '../Intro'
import { faq } from '../../constants/content/index'

type Props = {
  test: string
}

class FAQ extends Component<Props> {
  render () {
    return (
      <div className='FAQ' data-test={this.props.test}>
        <Helmet
          title='FAQ'
        />
        <Wrapper>
          <Navigation page='faq'/>
          <MainContentGrid>
            <header>
              <PageTitle {...faq}/>
              <Intro {...faq}/>
            </header>
          </MainContentGrid>
        </Wrapper>
      </div>
    )
  };
}

export default FAQ

const Wrapper = styled.div`
  display: flex;
`
