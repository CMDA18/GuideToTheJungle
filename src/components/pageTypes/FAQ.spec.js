import React from 'react'
import { shallow } from 'enzyme'
import FAQ from './FAQ'

it('renders without crashing', () => {
  expect(shallow(<FAQ />)).toMatchSnapshot()
})
