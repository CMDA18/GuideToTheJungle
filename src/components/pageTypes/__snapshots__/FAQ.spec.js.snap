// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FAQ />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
          title="FAQ"
        />,
        <FAQ__Wrapper>
          <Navigation
            page="faq"
          />
          <MainContentGrid>
            <header>
              <PageTitle
                introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                introTitle="Questions to ..."
                pageTitle="0.6 FAQ"
              />
              <Intro
                introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                introTitle="Questions to ..."
                pageTitle="0.6 FAQ"
              />
            </header>
          </MainContentGrid>
        </FAQ__Wrapper>,
      ],
      "className": "FAQ",
      "data-test": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defer": true,
          "encodeSpecialCharacters": true,
          "title": "FAQ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Navigation
              page="faq"
            />,
            <MainContentGrid>
              <header>
                <PageTitle
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="Questions to ..."
                  pageTitle="0.6 FAQ"
                />
                <Intro
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="Questions to ..."
                  pageTitle="0.6 FAQ"
                />
              </header>
            </MainContentGrid>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "page": "faq",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <header>
                <PageTitle
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="Questions to ..."
                  pageTitle="0.6 FAQ"
                />
                <Intro
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="Questions to ..."
                  pageTitle="0.6 FAQ"
                />
              </header>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <PageTitle
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="Questions to ..."
                    pageTitle="0.6 FAQ"
                  />,
                  <Intro
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="Questions to ..."
                    pageTitle="0.6 FAQ"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                    "introTitle": "Questions to ...",
                    "pageTitle": "0.6 FAQ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                    "introTitle": "Questions to ...",
                    "pageTitle": "0.6 FAQ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "header",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            title="FAQ"
          />,
          <FAQ__Wrapper>
            <Navigation
              page="faq"
            />
            <MainContentGrid>
              <header>
                <PageTitle
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="Questions to ..."
                  pageTitle="0.6 FAQ"
                />
                <Intro
                  introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                  introTitle="Questions to ..."
                  pageTitle="0.6 FAQ"
                />
              </header>
            </MainContentGrid>
          </FAQ__Wrapper>,
        ],
        "className": "FAQ",
        "data-test": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defer": true,
            "encodeSpecialCharacters": true,
            "title": "FAQ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Navigation
                page="faq"
              />,
              <MainContentGrid>
                <header>
                  <PageTitle
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="Questions to ..."
                    pageTitle="0.6 FAQ"
                  />
                  <Intro
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="Questions to ..."
                    pageTitle="0.6 FAQ"
                  />
                </header>
              </MainContentGrid>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "page": "faq",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <header>
                  <PageTitle
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="Questions to ..."
                    pageTitle="0.6 FAQ"
                  />
                  <Intro
                    introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                    introTitle="Questions to ..."
                    pageTitle="0.6 FAQ"
                  />
                </header>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <PageTitle
                      introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                      introTitle="Questions to ..."
                      pageTitle="0.6 FAQ"
                    />,
                    <Intro
                      introText="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi."
                      introTitle="Questions to ..."
                      pageTitle="0.6 FAQ"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                      "introTitle": "Questions to ...",
                      "pageTitle": "0.6 FAQ",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "introText": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur porta dolor ac sem egestas blandit. Praesent eget tempor nisi.",
                      "introTitle": "Questions to ...",
                      "pageTitle": "0.6 FAQ",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "header",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
